### PR TITLE
#61 added a test for pay operation

### PR DESCRIPTION
**Pull request by ammaratef45**
PR for #61 
 - added a test for pay operation
 - added an enhancement for the pay method

### CODE REVIEW

**Comment by ammaratef45** [comment]
@0crat status

**Comment by 0crat** [comment]
I'm not managing `amihaiemil/zold-java-client` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by ammaratef45** [comment]
@amihaiemil looks like you should link the project again

**Comment by ammaratef45** [comment]
@denis-konovalyenko can I ask you to review this PR? we have an issue with zerocrat currently and you won't get points.

File: src/test/java/com/amihaiemil/zold/RtWalletTestCase.java [hunk 1]
```diff
@@ -199,4 +199,38 @@ public void findsWalletTransactions() throws Exception {
...
     }
+
+    /**
+     * Checks whether the {@link RtWallet#pay(String, String, String, String)}
+     * invocation succeeds.
+     * @throws Exception If an exception is thrown
+     */
+    @Test
+    public void pays() throws Exception  {
+        final ClassicHttpResponse response = new Response(
```
Comment (denis-konovalyenko) [inline src/test/java/com/amihaiemil/zold/RtWalletTestCase.java]: @ammaratef45 ,  could you please check the indentation for the new code? It should have been 4 spaces I think.

File: src/test/java/com/amihaiemil/zold/RtWalletTestCase.java [hunk 2]
```diff
@@ -199,4 +199,38 @@ public void findsWalletTransactions() throws Exception {
...
+                HttpStatus.SC_OK,
+                ContentType.TEXT_PLAIN,
+                "12345"
+        );
+        final HttpClient httpClient = new MockHttpClient(
+                new AssertRequest(
+                        response,
+                        new Condition(
```
Comment (denis-konovalyenko) [inline src/test/java/com/amihaiemil/zold/RtWalletTestCase.java]: @ammaratef45 , could you please add another condition for checking the request payload?

**Comment by denis-konovalyenko** [comment]
@ammaratef45 , do you happen to know the exact issue with 0crat? I think my agenda is just full of other things right now... :)

**Comment by amihaiemil** [comment]
@ammaratef45 I asked Kirill on Telegram to resurect the project, he said it should be fixed now

**Comment by amihaiemil** [comment]
@0crat assign @denis-konovalyenko

**Comment by ammaratef45** [comment]
@amihaiemil yes but you still need to link it in slack again.

**Comment by 0crat** [comment]
I'm not managing `amihaiemil/zold-java-client` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by ammaratef45** [comment]
> @ammaratef45 , do you happen to know the exact issue with 0crat? I think my agenda is just full of other things right now... :)

@denis-konovalyenko 
beside that, he just thinks he isn't managing the project.

File: src/test/java/com/amihaiemil/zold/RtWalletTestCase.java (hunk 2 above)
Comment (ammaratef45) [reply src/test/java/com/amihaiemil/zold/RtWalletTestCase.java]: @denis-konovalyenko the pay method is void and payload isn't visible, can you explain how I can test it? Thanks

Comment (denis-konovalyenko) [reply src/test/java/com/amihaiemil/zold/RtWalletTestCase.java]: @ammaratef45 , try to use `com.amihaiemil.zold.PayloadOf#PayloadOf` wiith `com.amihaiemil.zold.mock.Condition'.

Comment (denis-konovalyenko) [reply src/test/java/com/amihaiemil/zold/RtWalletTestCase.java]: @ammaratef45 , I have made a related commit in my branch [here](https://github.com/denis-konovalyenko/zold-java-client/tree/61). Could you please take a look when have time?

Comment (denis-konovalyenko) [reply src/test/java/com/amihaiemil/zold/RtWalletTestCase.java]: @ammaratef45  ping :)

**Comment by denis-konovalyenko** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/amihaiemil/zold-java-client/issues/62#issuecomment-491400247))

@denis-konovalyenko This is what I know about this job in GFCHY7NQG, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#62](https://github.com/amihaiemil/zold-java-client/issues/62) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/GFCHY7NQG?q=%7Bjob%3A%27gh%3Aamihaiemil%2Fzold-java-client%2362%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

File: src/test/java/com/amihaiemil/zold/RtWalletTestCase.java (hunk 2 above)
Comment (ammaratef45) [reply src/test/java/com/amihaiemil/zold/RtWalletTestCase.java]: @denis-konovalyenko I forgot about this PR :)
I've merged your branch, Thanks for help

**Comment by ammaratef45** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/amihaiemil/zold-java-client/issues/62#issuecomment-491574130))

@ammaratef45 This is what I know about this job in GFCHY7NQG, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#62](https://github.com/amihaiemil/zold-java-client/issues/62) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/GFCHY7NQG?q=%7Bjob%3A%27gh%3Aamihaiemil%2Fzold-java-client%2362%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ammaratef45** [comment]
@0crat assign @denis-konovalyenko

**Comment by 0crat** [comment]
> @0crat assign @denis-konovalyenko ([here](https://github.com/amihaiemil/zold-java-client/issues/62#issuecomment-491574585))

@ammaratef45 You can't do "Request order start", unless you have one of these roles: `ARC`, `PO`; current roles of @ammaratef45[/z](https://www.0crat.com/u/ammaratef45) are: `DEV`, `REV`

File: src/test/java/com/amihaiemil/zold/RtWalletTestCase.java (hunk 2 above)
Comment (denis-konovalyenko) [reply src/test/java/com/amihaiemil/zold/RtWalletTestCase.java]: @ammaratef45 , thanks!

**Comment by denis-konovalyenko** [comment]
@ammaratef45 , @amihaiemil , I do not know whether we can just merge or assign this to me as a reviewer before the merge.

So, it is up to you.

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by ammaratef45** [comment]
@rultor merge it

**Comment by 0crat** [comment]
Job `gh:amihaiemil/zold-java-client#62` is not assigned, can't get performer

**Comment by 0crat** [comment]
This job is not in scope

**Pull request merged**